src/ln/channel.rs

```diff
@@ -1352,6 +1355,9 @@ impl Channel {
 				self.cur_local_commitment_transaction_number != INITIAL_COMMITMENT_NUMBER {
 			panic!("Should not have advanced channel commitment tx numbers prior to funding_created");
 		}
+		if self.channel_monitor.funding_txo.clone().unwrap().0.to_channel_id() != msg.channel_id {
+			return Err(HandleError{err: "Received incorrect channel_id in funding_signed message", action: None});
```

Comment:
Given how things are structured it'd be very hard for someone to introduce a bug that will make this condition false, but if that ever happens it should be hard panic, not returning Err.

Reply:
I agree that it is very likely that it col be caused by a bug. But the remote could accidentally or intentionally send the incorrect id.

Reply:
ChannelManager handles an incorrect channel_id and channel_id collision.